cdap-security/src/main/java/co/cask/cdap/security/server/KnoxJAASLoginService.java

```diff
@@ -289,4 +288,9 @@ public void logout(UserIdentity user) {
     }
   }
 
+  @Override
+  public String serverCaseConvert(String username) {
+    throw new UnsupportedOperationException("server type is not supported for KnoxLoginService");
```

Comment:
We can add implementation of serverCaseConvert for Knox as well. We can extract user name from Knox Token and return it.